### PR TITLE
Run3-alca247X Add possibility to uncorrect the RecHit energies for AlCaReco tests

### PR DESCRIPTION
**Pull request by bsunanda**
#### PR description:

Add possibility to uncorrect the RecHit energies for AlCaReco tests

#### PR validation:

Tested by creating the flat trees with uncorrected energies

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

Nothing special

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

File: Calibration/HcalCalibAlgos/plugins/HcalIsoTrackAnalyzer.cc [hunk 1]
```diff
@@ -324,6 +370,13 @@ void HcalIsoTrackAnalyzer::beginJob() {
 }
 
 // ------------ method called when starting to processes a run  ------------
+void HcalIsoTrackAnalyzer::beginRun(edm::Run const& iRun, edm::EventSetup const& iSetup) {
+  theHBHETopology_ = &iSetup.getData(tok_htopo_);
+  const HcalRespCorrs* resp = &iSetup.getData(tok_respcorr_);
+  respCorrs_ = new HcalRespCorrs(*resp);
```
Comment (perrotta) [inline Calibration/HcalCalibAlgos/plugins/HcalIsoTrackAnalyzer.cc:376]: Shouldn't this get deleted at the endRun?

Comment (bsunanda) [reply Calibration/HcalCalibAlgos/plugins/HcalIsoTrackAnalyzer.cc:376]: Done

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by perrotta** [comment]
+alca
- RelValInput error is unrelated

**Comment by rappoccio** [comment]
+1

**Comment by perrotta** [comment]
@rappoccio an explicit "merge" is needed here, because of the test failure

**Comment by rappoccio** [comment]
merge

**Pull request merged**